parser/src/parser/tests/evaluator_functions.rs

```diff
@@ -0,0 +1,162 @@
+
+#[test]
+fn ev_fn_main_cols() {
+    let source = "
+    ev advance_clock(main: [clk]):
+        enf clk' = clk + 1";
+    let expected = Source(vec![SourceSection::EvaluatorFunction(
+        EvaluatorFunction::new(
+            Identifier("advance_clock".to_string()),
+            vec![TraceCols::new(Identifier("clk".to_string()), 1)],
+            vec![],
+            vec![Constraint(IntegrityConstraint::new(
+                NamedTraceAccess(NamedTraceAccess::new(Identifier("clk".to_string()), 0, 1)),
+                Add(
+                    Box::new(Elem(Identifier("clk".to_string()))),
+                    Box::new(Const(1)),
+                ),
+            ))],
+        ),
+    )]);
+    build_parse_test!(source).expect_ast(expected);
+}
+
+#[test]
+fn ev_fn_main_and_aux_cols() {
+    let source = "
+    ev ev_func(main: [clk], aux: [a, b]):
+        let z = a + b
+        enf clk' = clk + 1
+        enf a' = a + z";
+
+    let expected = Source(vec![SourceSection::EvaluatorFunction(
+        EvaluatorFunction::new(
+            Identifier("ev_func".to_string()),
+            vec![TraceCols::new(Identifier("clk".to_string()), 1)],
+            vec![
+                TraceCols::new(Identifier("a".to_string()), 1),
+                TraceCols::new(Identifier("b".to_string()), 1),
+            ],
+            vec![
+                Variable(Variable::new(
+                    Identifier("z".to_string()),
+                    VariableType::Scalar(Add(
+                        Box::new(Elem(Identifier("a".to_string()))),
+                        Box::new(Elem(Identifier("b".to_string()))),
+                    )),
+                )),
+                Constraint(IntegrityConstraint::new(
+                    NamedTraceAccess(NamedTraceAccess::new(Identifier("clk".to_string()), 0, 1)),
+                    Add(
+                        Box::new(Elem(Identifier("clk".to_string()))),
+                        Box::new(Const(1)),
+                    ),
+                )),
+                Constraint(IntegrityConstraint::new(
+                    NamedTraceAccess(NamedTraceAccess::new(Identifier("a".to_string()), 0, 1)),
+                    Add(
+                        Box::new(Elem(Identifier("a".to_string()))),
+                        Box::new(Elem(Identifier("z".to_string()))),
+                    ),
+                )),
+            ],
+        ),
+    )]);
+    build_parse_test!(source).expect_ast(expected);
+}
+
+#[test]
+fn ev_fn_call() {
+    let source = "
+    integrity_constraints:
+        enf advance_clock([clk])";
+
+    let expected = Source(vec![SourceSection::IntegrityConstraints(vec![
+        EvaluatorFunctionCall(EvaluatorFunctionCall::new(
+            Identifier("advance_clock".to_string()),
+            vec![vec![TraceCols::new(Identifier("clk".to_string()), 1)]],
+        )),
+    ])]);
+
+    build_parse_test!(source).expect_ast(expected);
+}
+
+#[test]
+fn ev_fun_call_inside_ev_fn() {
```

Comment:
nit: I would probably do `ev_fn_call_inside_ev_fn` for consistency with our other names